include/pybind11/pybind11.h

```diff
@@ -2751,10 +2751,15 @@ get_type_override(const void *this_ptr, const type_info *this_type, const char *
         if ((std::string) str(f_code->co_name) == name && f_code->co_argcount > 0) {
             PyObject *locals = PyEval_GetLocals();
             if (locals != nullptr) {
+#        if PY_VERSION_HEX >= 0x030b0000
+                PyObject *co_varnames = PyCode_GetVarnames(f_code);
+#        else
                 PyObject *co_varnames = PyObject_GetAttrString((PyObject *) f_code, "co_varnames");
+#        endif
                 PyObject *self_arg = PyTuple_GET_ITEM(co_varnames, 0);
                 Py_DECREF(co_varnames);
                 PyObject *self_caller = dict_getitem(locals, self_arg);
+                Py_DECREF(locals);
```

Comment:
Surprised we missed this.

Reply:
Unfortunately this turns out to be incorrect.

I only discovered that when trying to deploy this PR at Google, but in retrospect it's totally clear:

https://docs.python.org/3/c-api/reflection.html#c.PyEval_GetLocals

> Return value: Borrowed reference. Part of the [Stable ABI](https://docs.python.org/3/c-api/stable.html#stable).

I'll take care of removing this line again. (I'll report the full error message there.)

Reply:
That's https://github.com/pybind/pybind11/pull/4927 now.

Reply:
Oh, I didn't notice the CPython API convention.